### PR TITLE
Decode PATH_INFO in router_rewrite plugin before putting it in the wsgi environment, per wsgi spec

### PR DESCRIPTION
**Pull request by bhearsum**
Background on this in #1263.

### CODE REVIEW

**Comment by bhearsum** [comment]
@unbit - does seem sensible based on https://github.com/unbit/uwsgi/issues/1263?

**Comment by unbit** [comment]
Just noted now you fixed the leak, yes sorry for the late. The patch looks good. @xrmx when you have time backport it to stable, thanks a lot

File: plugins/router_rewrite/router_rewrite.c [hunk 1]
```diff
@@ -38,9 +38,14 @@ static int uwsgi_routing_func_rewrite(struct wsgi_request *wsgi_req, struct uwsg
 		}
 	}
 
-	char *ptr = uwsgi_req_append(wsgi_req, "PATH_INFO", 9, ub->buf, path_info_len);
+	char *path_info = uwsgi_malloc(path_info_len);
+	http_url_decode(ub->buf, &path_info_len, path_info);
+
+	char *ptr = uwsgi_req_append(wsgi_req, "PATH_INFO", 9, path_info, path_info_len);
         if (!ptr) goto clear;
```
Comment (catlee) [inline plugins/router_rewrite/router_rewrite.c]: do you need to free path_info in case uswgi_req_append fails?

Comment (bhearsum) [inline plugins/router_rewrite/router_rewrite.c]: Yeah. A couple of others pointed this out too :). https://github.com/unbit/uwsgi/pull/1284 should fix it.

**Comment by xrmx** [comment]
Fixes folded and push to uwsgi-2.0 as 18a593b8f5cf6c8c87affb2ef3fbf22e865d4cbb

**Pull request merged**